### ISSUE TITLE
Duplicate entries in "Recent Projects" and "Home page"

### ISSUE THREAD

**Issue by piotrzarzycki21** [Moonshine-IDE/Moonshine-IDE#1230]
In menu **File** -> **Recent Projects** and **Home page** I have found duplicate entries. In order to reproduce it I believe you need to open several times one project - maybe in a different workspaces as well. 

<img width="1575" alt="Screenshot 2023-11-28 at 11 05 36" src="https://github.com/Moonshine-IDE/Moonshine-IDE/assets/24554795/3eab06cf-a834-4289-b9c9-49cd1fb2c8a0">

Moonshine: 3.3.4 build 3105. Running on: Mac OS 14.1.1 (Sonoma)

**Comment by piotrzarzycki21** [comment]
Please @rat-moonshine take a look into that. It's been annoying for a long time now.

**Comment by rat-moonshine** [comment]
Are you seeing false click also, when trying to open project from home screen?

**Comment by piotrzarzycki21** [comment]
> Are you seeing false click also, when trying to open project from home screen?

Yes!!

**Comment by JoelProminic** [comment]
I am also seeing some duplicate projects.  Checking the projects with this problem, I'm seeing that they are all cases where the actual project name (determined by the proj file) does not match the directory name.  @piotrzarzycki21, does this match what you are seeing as well.

Currently, none of the RECENT entries on the Home page are opening for me, so I think this may be a separate issue.

**Comment by piotrzarzycki21** [comment]
> @piotrzarzycki21, does this match what you are seeing as well.

Yes It is exactly like that. It maybe a good hint to resolve that issue.

**Comment by rat-moonshine** [comment]
Updates:

- Fixed scenario where newly opened project do not adds in the recent project-list in a fresh Moonshine installation 
- Fixed duplication of items in recent project-list and in so object

Note:
In case of duplicated items those already saved in the SO, and since I don't have a good way to determine which item was suppose to be displayed and which item duplicated with a different name (unless additional process to determine which I feel costly at this point), it's problematic to choose between. Thus, I'm adding here a force recent-project-list clean from SO, which will run once, and therefore newly updated list shouldn't have any further duplicated entry. 

A new build should be ready soon.

**Comment by piotrzarzycki21** [comment]
> Thus, I'm adding here a force recent-project-list clean from SO, which will run once, and therefore newly updated list shouldn't have any further duplicated entry.

I'm not sure if I'm in favour of that part. We can do clear that list by reseting settings - this is the way. Adding logic which works once, but what that means ? It will be initiated once per Moonshine launch ?

**Comment by rat-moonshine** [comment]
My reasoning to not to educate user separately if this scenario already exists to someone, and s/he don't know how to fix this. Settings reset is always there anyway. The once force-clean logic will trigger only once after you install new build, and it'll clean only the recent-project-list field from the sol file. 

If we still want user to find-out how to fix an already messed-up list, then I can stop the logic.

**Comment by piotrzarzycki21** [comment]
It's working in my tests, so I'm ok with this fix. Could you also look into "Recent Files"? - it seems to be same issue.

**Comment by piotrzarzycki21** [comment]
When I'm switching between workspaces issue can be still reproducible. I see duplication in recent. When you will be able look into that @rat-moonshine

<img width="1305" alt="Screenshot 2023-12-22 at 10 09 15" src="https://github.com/Moonshine-IDE/Moonshine-IDE/assets/24554795/98740399-c73c-44c9-ad0f-8cbd879a0b37">

**Comment by rat-moonshine** [comment]
I tried to import different same projects in multiple workspaces, for some time, I couldn't able to reproduce the problem yet, unfortunately. 

If you would have any idea on how to reproduce one @piotrzarzycki21 , please mention. Meanwhile, I'm still looking if I can able to reproduce the problem.

**Comment by JoelProminic** [comment]
While trying to reproduce @piotrzarzycki21's error, I got an error when opening a workspace with a Domino Visual Editor project:


    : Error #1009
    : TypeError: Error #1009
    : 	at spark.layouts.supportClasses::NavigatorLayoutBase/updateElementsInLayout()
    : 	at spark.layouts.supportClasses::NavigatorLayoutBase/updateDisplayList()
    : 	at spark.layouts::StackLayout/updateDisplayList()
    : 	at spark.components.supportClasses::GroupBase/updateDisplayList()
    : 	at spark.containers::DeferredGroup/updateDisplayList()
    : 	at mx.core::UIComponent/validateDisplayList()
    : 	at mx.managers::LayoutManager/validateDisplayList()
    : 	at mx.managers::LayoutManager/doPhasedInstantiation()
    : 	at mx.managers::LayoutManager/doPhasedInstantiationCallback()

The RECENT section stopped displaying, but it reappeared when I put my mouse over the expected list and scrolled.

**Comment by JoelProminic** [comment]
I am also seeing duplicate entries for the MyAccount project (Note that the `app` projects are actually different directories):
![image](https://github.com/Moonshine-IDE/Moonshine-IDE/assets/24532731/641aa77f-26c2-4893-bd41-4e2a6eb275e5)

To trigger this, I am switching between my MyAccount workspace and other repositories with **different** projects, and then back to the MyAccount workspace.  After doing this a few times, I see duplicate MyAccount entries.

**Comment by JoelProminic** [comment]
I'm still thinking this is related to cases where the project name does not match the directory name. Steps to reproduce:
1. Create a new Royale Browser Project (Jewel) called:  TestRoyaleDirectory
2. Right-click > Show in Finder
3. Close the project
4. Rename TestRoyaleDirecotry.as3proj to TestRoyaleProject.as3proj
5. Double-click TestRoyaleProject.as3proj to open it in Moonshine.  The project entry is "TestRoyaleProject", while the RECENT name is TestRoyaleDirectory.  At this point, I already see duplicate entries in RECENT - are these RECENT entries based on the project file or the directory?
6. Open a different workspace
7. Open the previous workspace.  BUG:  Another TestRoyaleDirectory entry appears in RECENTS

**Comment by rat-moonshine** [comment]
> I'm still thinking this is related to cases where the project name does not match the directory name. Steps to reproduce:
> 
> 1. Create a new Royale Browser Project (Jewel) called:  TestRoyaleDirectory
> 2. Right-click > Show in Finder
> 3. Close the project
> 4. Rename TestRoyaleDirecotry.as3proj to TestRoyaleProject.as3proj
> 5. Double-click TestRoyaleProject.as3proj to open it in Moonshine.  The project entry is "TestRoyaleProject", while the RECENT name is TestRoyaleDirectory.  At this point, I already see duplicate entries in RECENT - are these RECENT entries based on the project file or the directory?
> 6. Open a different workspace
> 7. Open the previous workspace.  BUG:  Another TestRoyaleDirectory entry appears in RECENTS

It is correct that the opening title in sidebar should reflect in recent-list, as well. I see my last changes affected to that and the recent-items were displaying the directory name, instead of the project name that being shown in sidebar.

I again did one experimental change, but I see it's becoming hard for me to easily reproduce the scenario again; @piotrzarzycki21 please give some tests with newer build, if you're still seeing the problem after couple of days' works.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine can you check on bamboo if build is working for Mac and Win ?

**Comment by rat-moonshine** [comment]
> @rat-moonshine can you check on bamboo if build is working for Mac and Win ?

Yeah, I see I missed to commit a change from MoonshineGUICore. It's now committed and a build starts soon.

**Comment by piotrzarzycki21** [comment]
It looks like it's fixed @rat-moonshine

**Comment by rat-moonshine** [comment]
> I did trough menu Settings -> General -> Reset to Default - I believe it should remove all entries in Recent, but instead it does nothing to that area.

I can check that.

**Comment by JoelProminic** [comment]
I couldn't reproduce the duplicates in my tests today.
